python/cusignal/_spectral.py

```diff
@@ -270,14 +271,18 @@ def _get_backend_kernel(dtype, grid, block, stream, use_numba):
     )
 
 
-def _populate_kernel_cache():
-    for numba_type, c_type in _SUPPORTED_TYPES.items():
-        # JIT compile the numba kernels
-        sig = _numba_lombscargle_signature(numba_type)
-        _numba_kernel_cache[(str(numba_type))] = cuda.jit(sig, fastmath=True)(
-            _numba_lombscargle
-        )
+def _populate_kernel_cache(np_type, use_numba):
+
+    try:
+        numba_type, c_type = _SUPPORTED_TYPES.get(np_type)
+    except TypeError:
+        print("Data type {} not supported.".format(np_type))
+        return
```

Comment:
If the user tries to precompile the kernel for an unsupported data type, then we still return (albeit with the error message printed out) - would it be less surprising to instead propagate an error back to the user (e.g. raising an exception out of here)? If one misses the print message, it could lead to the impression that precompilation was successful for the given unsupported dtype, when instead the kernel will always run with a `float64` data type (from my reading of spectral.py).

Reply:
I believe my implementing `__getitem__` and `KeyError` this issue goes away.